### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by sixlettervariables**
Adds XML documentation.

### CODE REVIEW

File: NATS.Client/AsyncSub.cs [hunk 1]
```diff
@@ -3,13 +3,22 @@
 using System;
 using System.Threading.Tasks;
 
-// disable XML comment warnings
-#pragma warning disable 1591
-
 namespace NATS.Client
 {
+    /// <summary>
+    /// <see cref="AsyncSubscription"/> asynchronously delivers messages to listeners of the <see cref="MessageHandler"/>
```
Comment (ColinSullivan1) [inline NATS.Client/AsyncSub.cs:9]: Can you add a note that this class shouldn't be used directly - to instead use the interface.

File: NATS.Client/Conn.cs [hunk 2]
```diff
@@ -13,20 +13,38 @@
...
-// disable XML comment warnings
-#pragma warning disable 1591
-
 namespace NATS.Client
 {
     /// <summary>
-    /// State of the connection.
+    /// State of the <see cref="IConnection"/>.
     /// </summary>
     public enum ConnState
```
Comment (ColinSullivan1) [inline NATS.Client/Conn.cs:21]: Thanks, I missed these!

File: NATS.Client/Subscription.cs [hunk 3]
```diff
@@ -3,11 +3,11 @@
...
 
-// disable XML comment warnings
-#pragma warning disable 1591
-
 namespace NATS.Client
 {
+    /// <summary>
+    /// Represents interest in a NATS topic.
+    /// </summary>
     public class Subscription : ISubscription, IDisposable
```
Comment (ColinSullivan1) [inline NATS.Client/Subscription.cs]: Maybe add a comment to use ISubscription instead?

**Review by ColinSullivan1 — CHANGES_REQUESTED** [review]
Great work here, really polishes up the doc.  Much appreciated!

**Comment by ColinSullivan1** [comment]
BTW - Doxygen found some issues; these were from running  build_doc.bat in the `<src>/documentation` directory.  I've attached output.

[doxyout.txt](https://github.com/sixlettervariables/csnats/files/1136027/doxyout.txt)

**Comment by sixlettervariables** [comment]
It appears doxygen does not handle `<see langword="">`. I'll replace these with something it supports.

File: NATS.Client/Subscription.cs [hunk 4]
```diff
@@ -262,12 +306,21 @@ protected virtual void Dispose(bool disposing)
...
         }
 
+        /// <summary>
+        /// Releases all resources used by the <see cref="Subscription"/>.
+        /// </summary>
+        /// <remarks>This method unsubscribes from the subject, to release resources.</remarks>
         public void Dispose()
         {
             Dispose(true);
+            GC.SuppressFinalize(this);
```
Comment (ColinSullivan1) [inline NATS.Client/Subscription.cs:316]: +1

**Review by ColinSullivan1 — APPROVED** [review]
I've spot checked these, and they look great.  Great addition to the NATS client.  Thanks for all of the work!

**Pull request merged**